File: BOJ/1769. 3의 배수/solution.js
const n = require('fs').readFileSync('/dev/stdin').toString().trim();
let sum = [...n.toString()], cnt = 0;
while (sum.length > 1) {
    sum = Array.prototype.reduce.call(sum, (acc, cur) => acc + (+cur), 0).toString();
    cnt += 1;
}
console.log(cnt);
console.log(+sum % 3 === 0 ? 'YES' : 'NO');

File: BOJ/1769. 3의 배수/solution.cc
#include <iostream>

using namespace std;

char nums[1000001];

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr);
    cout.tie(nullptr);
    cin >> nums;

    int i = 0, n = 0;
    for (i; nums[i]; i += 1) n += nums[i] - '0';
    int cnt = i > 1 ? 1 : 0;

    while (n > 9) {
        int sum = 0;
        while (n) {
            sum += n % 10;
            n /= 10;
        }
        cnt += 1;
        n = sum;
    }

    cout << cnt << '\n';
    cout << (n % 3 == 0 ? "YES" : "NO");
    return 0;
}